test.js

```diff
@@ -29,35 +29,86 @@ const format = async (format, code) => {
 
 // https://developer.mozilla.org/en-US/docs/Glossary/Void_element
 const allVoidElements = [
-  "area",
-  "base",
-  "br",
-  "col",
-  "embed",
-  "hr",
-  "img",
-  "input",
-  "link",
-  "meta",
-  "param",
-  "source",
-  "track",
-  "wbr",
+  { el: "area", hasTrailingNewline: false },
+  { el: "base", hasTrailingNewline: false },
+  { el: "br", hasTrailingNewline: false },
+  { el: "col", hasTrailingNewline: true },
+  { el: "embed", hasTrailingNewline: false },
+  { el: "hr", hasTrailingNewline: true },
+  { el: "img", hasTrailingNewline: false },
+  { el: "input", hasTrailingNewline: false },
+  { el: "link", hasTrailingNewline: false },
+  { el: "meta", hasTrailingNewline: false },
+  { el: "param", hasTrailingNewline: true },
+  { el: "source", hasTrailingNewline: true },
+  { el: "track", hasTrailingNewline: true },
+  { el: "wbr", hasTrailingNewline: false },
```

Comment:
How is `hasTrailingNewline` determined for each tag? Is this defined in the Prettier source somewhere?

Reply:
Tbh I don't remember how I figured that out, but there's a decent probability it was just through running prettier without the plugin and observing the results.

I think the answer lies in [printChildren](https://github.com/prettier/prettier/blob/06ece395beb760f8b8b62eb4a46284cb13d82508/src/language-html/print/children.js#L136)'s call to [printBetweenLine](https://github.com/prettier/prettier/blob/06ece395beb760f8b8b62eb4a46284cb13d82508/src/language-html/print/children.js#L74) but I've failed to trace it through to that conclusion.

Reply:
Okay, no worries. I'll just add a comment here saying something like that